library/Rules/Uuid.php

```diff
@@ -13,19 +13,78 @@
 
 namespace Respect\Validation\Rules;
 
+use Respect\Validation\Exceptions\ComponentException;
 use function is_scalar;
 use function preg_match;
 
+/**
+ * This class validates UUIDs.
+ *
+ * Henrique Moody <henriquemoody@gmail.com>
+ * Michael Weimann <mail@michael-weimann.eu>
+ */
 class Uuid extends AbstractRule
 {
-    private const PATTERN = '/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
+    public const VERSION_ALL = '[1-5]';
+    public const VERSION_1 = '1';
+    public const VERSION_2 = '2';
+    public const VERSION_3 = '3';
+    public const VERSION_4 = '4';
+    public const VERSION_5 = '5';
+
+    public const VERSIONS = [
+        self::VERSION_ALL,
+        self::VERSION_1,
+        self::VERSION_2,
```

Comment:
For what's seem the version 2 is a bit more complicated than that: http://www.nesubl.eu/data/guidelines/NES%20Guideline%20-%20UUID%20-%20Version%202.pdf, one example that can be found in this documentation is `6E09886B-DC6E-439F-82D1-7CCAC7F4E3B3`, which wouldn't pass this validation.

I think would be better to simply drop the version 2 for now.